mlperf_logging/result_summarizer/result_summarizer.py

```diff
@@ -282,21 +302,30 @@ def summarize_results(folder, ruleset):
             allowed_benchmarks = _ALLOWED_BENCHMARKS_V07
         elif ruleset == '1.0.0':
             allowed_benchmarks = _ALLOWED_BENCHMARKS_V10
+        csv_header.extend(allowed_benchmarks)
         for benchmark in allowed_benchmarks:
             if benchmark in benchmark_scores:
                 row += '{:.2f},'.format(benchmark_scores[benchmark])
             else:
                 row += ','
 
         # Construct postfix portion of the row.
+        csv_header.append('details_url')
         row += '{},'.format(_details_url(desc, ruleset))
+        csv_header.append('code_url')
         row += '{},'.format(_code_url(desc, ruleset))
 
         rows[_row_key(desc)] = row
 
     # Print rows in order of the sorted keys.
     for key in sorted(rows):
         print(rows[key])
+        # Add rows to csv object
+        if csv_obj is not None:
+            # Add the header above the first results row
+            if csv_obj.csv_rows == list():
+                csv_obj.add_row(csv_header)
+            csv_obj.add_row([*csv.reader([rows[key]])][0])
```

Comment:
It seems complicated. Maybe `rows[key].split(',')`?

Reply:
I agree,

However, the reason I did not use this due to the potential for commas inside the field data (embedded delimiters). This would result in an incorrect splitting of the rows into too many columns, hence why I used csv.reader to utilise the double quote text qualifiers that are already inserted into the fields.

The annoyance was that I could not see how to return 1 list from the expansion of the iterator. It returned a list within a list and so I used [0] to overcome this.